rpcs3/Emu/Cell/Modules/cellCamera.cpp

```diff
@@ -115,6 +116,40 @@ static bool check_dev_num(s32 dev_num)
 	return dev_num == 0;
 }
 
+template <typename F = std::function<bool()>> requires (std::is_convertible_v<std::invoke_result_t<F>, bool>)
+static error_code check_errors(camera_state state, camera_state to_check = camera_state::uninit, F&& func = [](){ return false; })
+{
+	if (state == camera_state::uninit)
+	{
+		return CELL_CAMERA_ERROR_NOT_INIT;
+	}
+
+	if (state >= to_check)
+	{
+		return CELL_OK;
+	}
+
+	if (func())
+	{
```

Comment:
I'm not a fan of what you're doing here tbh.
1. It never reaches this point if the camera state is correct. So it is only checked if the camera state is wrong anyway.
2. It is kinda spaghetti code and hard to read in the flow of the functions
3. It adds another abstraction that you need to wrap your head around if you try to match assembly to code for example

So I personally would prefer if this function stayed a state check, and the param checks would remain conventional.

Reply:
Oh, I see the ordering bug, will fix.
Although it packs common logic which in return reduces amount of LOC significantly. IMO it should be better documented as an optional ""optimization"" for code size. When this optimization cannot be followed one shall  return to the manual coding of checks.

Reply:
I appreciate the attempt to reduce LOC.
But i think in this case it's not really necessary, especially since it meddles with history as well

Reply:
And it makes this PR harder to review as well 💩

Reply:
The main reason I'm modifying it in the first place is to remove unneeded variables for savestates (better to remove before PR is merged), history will be modified either way.
Everything in addition is a bonus.

Reply:
Hmm maybe I can unwrap it relatively easily.